app/src/main/java/com/example/advanceduiapp/ui/pager/PagerFragment.kt

```diff
@@ -0,0 +1,65 @@
+package com.example.advanceduiapp.ui.pager
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.os.bundleOf
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.activityViewModels
+import com.example.advanceduiapp.databinding.FragmentPagerBinding
+
+class PagerFragment : Fragment() {
+    private val model: PagerViewModel by activityViewModels()
+
+    private var _binding: FragmentPagerBinding? = null
+    private val binding get() = _binding!!
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        _binding = FragmentPagerBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        with(binding){
+            textSlideshow.text = requireArguments().get(titleTag) as String
+            pagerImageView.setImageResource(requireArguments().get(imageResourceTag) as Int)
+        }
+
+        if (requireArguments().get(showButtonTag) as Boolean) {
+            binding.button.visibility = View.VISIBLE
+            binding.button.setOnClickListener {
+                model.showMultipleViewRecyclerFragment.value = true
```

Comment:
create a method in viewModel to change the state of the showMultipleViewRecyclerFragment